test/unit/sql/get-where-conditions.test.js

```diff
@@ -11,4 +11,16 @@ describe('QueryGenerator#getWhereConditions', () => {
       queryGenerator.getWhereConditions(new Date(), User.getTableName(), User);
     }).to.throw('Unsupported where option value');
   });
+
+  it('ignores undefined', () => {
+    const User = sequelize.define('User');
+
+    expect(queryGenerator.getWhereConditions(undefined, User.getTableName(), User)).to.eq('');
+  });
+
+  it('ignores null', () => {
+    const User = sequelize.define('User');
+
+    expect(queryGenerator.getWhereConditions(undefined, User.getTableName(), User)).to.eq('');
```

Comment:
Shouldn't this use `null`?

Reply:
If you mean that the part that uses `queryGenerator.getWhereConditions(undefined, ...)` should be `queryGenerator.getWhereConditions(null, ...)` then actually it doesn't make a difference because we are checking `null-ish` [values like this](https://github.com/sequelize/sequelize/blob/v6/src/dialects/abstract/query-generator.js#L2829) which in turn defines `no-op` for both `undefined` and `null` [here like this](https://github.com/sequelize/sequelize/blob/13f2e89f8b6147897e3e43f01487de51aebcde87/src/dialects/abstract/query-generator.js#L2343).

Reply:
Well no it should use null since the test is about null

Reply:
lol guess nobody noticed that!

Reply:
Good catch! Fixed in #15705